src/ReverseProxy/Health/ActiveHealthCheckMonitor.cs

```diff
@@ -167,19 +174,28 @@ private async Task<DestinationProbingResult> ProbeDestinationAsync(ClusterState
             return new DestinationProbingResult(destination, null, ex);
         }
 
+        var probeActivity = Observability.YarpActivitySource.StartActivity("Proxy destination health check", ActivityKind.Client);
```

Comment:
```suggestion
        using var probeActivity = Observability.YarpActivitySource.StartActivity("Proxy destination health check", ActivityKind.Client);
```
And then you can remove the calls to `Stop`